src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdaterTest.java

```diff
@@ -0,0 +1,265 @@
+            anyString(),
+            anyBoolean(),
+            any(CaseDataContent.class)
+        )).willReturn(CaseDetails.builder().id(1L).build());
+        given(exceptionRecordFinalizer.finalizeExceptionRecord(anyMap(), anyLong())).willReturn(originalFields);
+
+        // when
+        ProcessResult res = ccdCaseUpdater.updateCase(
+            exceptionRecord,
+            configItem,
+            true,
+            "idamToken",
+            "userId",
+            existingCase
+        );
+
+        // then
+        assertThat(res.getErrors()).isEmpty();
+        assertThat(res.getWarnings()).isEmpty();
+        assertThat(res.getExceptionRecordData()).isEqualTo(originalFields);
+    }
+
+    @Test
+    void should_update_case_handle_feign_exception() throws Exception {
+        // given
+        given(configItem.getService()).willReturn("Service");
+        given(existingCase.getId()).willReturn(1L);
+        given(existingCase.getCaseTypeId()).willReturn("caseTypeId");
+        given(configItem.getUpdateUrl()).willReturn("url");
+        given(authTokenGenerator.generate()).willReturn("token");
+        caseUpdateDetails = new CaseUpdateDetails("event_id", new HashMap<String, String>());
+        successfulUpdateResponse = new SuccessfulUpdateResponse(caseUpdateDetails, new ArrayList<>());
+        exceptionRecord = new ExceptionRecord(
+            "1",
+            "caseTypeId",
+            "12345",
+            "some jurisdiction",
+            SUPPLEMENTARY_EVIDENCE_WITH_OCR,
+            "Form1",
+            now(),
+            now(),
+            emptyList(),
+            emptyList()
+        );
+        given(caseUpdateClient.updateCase("url", existingCase, exceptionRecord, "token"))
+            .willReturn(successfulUpdateResponse);
+        given(eventResponse.getEventId()).willReturn("eventId");
+        given(eventResponse.getToken()).willReturn("token");
+        given(coreCaseDataApi.startForCaseworker(
+            anyString(),
+            anyString(),
+            anyString(),
+            anyString(),
+            anyString(),
+            anyString()))
+            .willReturn(eventResponse);
+        given(coreCaseDataApi.submitForCaseworker(
+            anyString(),
+            anyString(),
+            anyString(),
+            anyString(),
+            anyString(),
+            anyBoolean(),
+            any(CaseDataContent.class)
+        )).willThrow(new FeignException.BadRequest("Msg", "Body".getBytes()));
+
+        // when
+        CallbackException callbackException = catchThrowableOfType(() ->
+            ccdCaseUpdater.updateCase(
+                exceptionRecord,
+                configItem,
+                true,
+                "idamToken",
+                "userId",
+                existingCase
+            ),
+            CallbackException.class
+        );
+
+        // then
+        assertThat(callbackException.getMessage()).isEqualTo("Failed to update case");
+    }
+
+    @Test
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/HMCTS/bulk-scan-orchestrator/pullRequest?prid=4577310)